pkg/engine/internal/executor/project.go

```diff
@@ -12,42 +12,72 @@ import (
 )
 
 func NewProjectPipeline(input Pipeline, columns []physical.ColumnExpression, evaluator *expressionEvaluator) (*GenericPipeline, error) {
-	// Get the column names from the projection expressions
-	columnNames := make([]string, len(columns))
-
-	for i, col := range columns {
-		if colExpr, ok := col.(*physical.ColumnExpr); ok {
-			columnNames[i] = colExpr.Ref.Column
-		} else {
-			return nil, fmt.Errorf("projection column %d is not a column expression", i)
-		}
-	}
```

Comment:
it looks like this was doing some kind of memoization of the names, which I assume was for memory purposes. this became kind of confusing with this PR as it was only used in the one logical branch, which I assumed could just reference `col.Ref.Column` directly, so I removed it, but am happy to bring it back if someone can help me understand it's purpose a bit better.